crates/matrix-sdk/src/room/futures.rs

```diff
@@ -299,7 +299,10 @@ impl<'a> IntoFuture for SendAttachment<'a> {
                             ImageError::ThumbnailBiggerThanOriginal
                             | ImageError::FormatNotSupported,
                         ) => None,
-                        Err(error) => return Err(error.into()),
+                        Err(error) => {
+                            tracing::error!("Failed to generate thumbnail: {error}");
+                            None
+                        }
```

Comment:
I like this; can we remove the previous match arm that handles two `ImageError` variants, so they're also logged?

Also, can we use `warn!` instead of `error!`, since these are non-critical?

Reply:
I am not sure that `ThumbnailBiggerThanOriginal` should be logged, or at least not at the `warn` level.

It could allow the users to be lazy and always use `generate_thumbnail` without having to check the image's size. It is more an optimization of the code than a real error.

Reply:
Ok. Would it make sense to log the other error at the warn level, though?

Reply:
Yes, definitely.

Reply:
Changed